### PR TITLE
[Cisco ASA] Fix parsing issue for 313005 #5946

### PR DESCRIPTION
**Pull request by daixque**
## What does this PR do?

Address a parsing issue for [Cisco ASA 313005](https://www.cisco.com/c/en/us/td/docs/security/asa/syslog/b_syslog/syslog-messages-302003-to-342008.html#con_4771130).

If log line below is given:

```
No matching connection for ICMP error message: icmp src inside:192.168.2.2(LOCAL\xxx\yyy) dst inside:192.168.2.3 (type 3, code 3) on inside interface. Original IP payload: udp src 192.168.2.2/53 dst 192.168.2.3/10872.
```

Many of the fields (such as `source.ip`, `destination.ip`, etc.) are not extracted properly in current implementation. So this PR adds more proper processor for this log type.

## Related issues

Relates #5946

### CODE REVIEW

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -601,11 +601,18 @@ processors:
...
       description: "313004"
       pattern: "%{event.outcome} %{network.transport} type=%{_temp_.cisco.icmp_type}, from%{}addr %{source.address} on interface %{_temp_.cisco.source_interface} to %{destination.address}: no matching session"
-  - dissect:
+  - grok:
       if: "ctx._temp_.cisco.message_id == '313005'"
       field: "message"
       description: "313005"
-      pattern: "No matching connection for %{network.transport} error message: %{} on %{_temp_.cisco.source_interface} interface.%{}riginal IP payload: %{}"
+      patterns:
+        - "No matching connection for ICMP error message: %{NOTSPACE:network.transport} src %{NOTCOLON:_temp_.cisco.source_interface}:%{IPORHOST:source.ip} dst %{NOTCOLON:_temp_.cisco.destination_interface}:%{IPORHOST:destination.ip} \\(type %{NUMBER:_temp_.cisco.icmp_type}, code %{NUMBER:_temp_.cisco.icmp_code}\\) on %{NOTCOLON} interface.%{SPACE}Original IP payload: %{NOTSPACE:input.type} src %{IPORHOST:source.ip}/%{NUMBER:source.port} dst %{IPORHOST:destination.ip}/%{NUMBER:destination.port}."
```
Comment (bhapas) [inline packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:609]: There is a bug fix in Grok Processor in ElasticSearch 8.7 which results in duplicate values in the output fields

For example, in this case

`destination.ip: ["192.168.2.2,"192.168.2.2"]` . This is currently aligned with how Logstash does the grokking. So, As for now the pipelines should have a fix for this.

Refer to this https://github.com/elastic/beats/issues/35133#issuecomment-1517354203

**Review by bhapas — COMMENTED** [review]
Once the CI is done for the new commit , Can we manually trigger CI testing of this PR and set the stack parameter to 8.7.1-SNAPSHOT (https://fleet-ci.elastic.co/job/Ingest-manager/job/integrations/job/PR-5910/build?delay=0sec) just to make sure the integration pipeline works with the latest changes

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (daixque) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: @bhapas Thanks. I removed the duplicated field mapping expressions.

Comment (bhapas) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: Can we run a CI build against `8.7.1-SNAPSHOT` using this direct build
https://fleet-ci.elastic.co/job/Ingest-manager/job/integrations/job/PR-5948/build/delay=10s

Parameter being `8.7.1-SNAPSHOT`

Comment (daixque) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: @bhapas Done.
https://fleet-ci.elastic.co/job/Ingest-manager/job/integrations/job/PR-5948/4/

**Review by bhapas — APPROVED** [review]
LGTM

**Pull request merged**